### PR TITLE
Event subscribers info

### LINKED ISSUE #4486: Add event subscribers info to getAllEvents and getEvent controller
Add event subscribers info to getAllEvents and getEvent controller to see if current user has already joined to event or not

### PR DESCRIPTION
**Pull request by danylohlynsky**
Issue: #4486

### CODE REVIEW

**Review by Max877866 — APPROVED** [review]

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 1]
```diff
@@ -97,16 +98,36 @@ public void delete(Long eventId, String email) {
     }
 
     @Override
-    public EventDto getEvent(Long eventId) {
+    public EventDto getEvent(Long eventId, Principal principal) {
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/EventServiceImpl.java:101]: Try to use here OptionalApi()

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 2]
```diff
@@ -97,16 +98,36 @@ public void delete(Long eventId, String email) {
...
         Event event =
             eventRepo.findById(eventId).orElseThrow(() -> new NotFoundException(ErrorMessage.EVENT_NOT_FOUND));
-        return modelMapper.map(event, EventDto.class);
+        EventDto eventDto = modelMapper.map(event, EventDto.class);
+        if (principal != null) {
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/EventServiceImpl.java:105]: Try to use here OptionalApi()

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 3]
```diff
@@ -97,16 +98,36 @@ public void delete(Long eventId, String email) {
...
     }
 
     @Override
-    public PageableAdvancedDto<EventDto> getAll(Pageable page) {
-        Page<Event> pages = eventRepo.findAllByOrderByIdDesc(page);
-        return buildPageableAdvancedDto(pages);
+    public PageableAdvancedDto<EventDto> getAll(Pageable page, Principal principal) {
+        Page<Event> events = eventRepo.findAllByOrderByIdDesc(page);
+        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(events);
+        if (principal != null) {
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/EventServiceImpl.java:117]: !

**Review by Sakhnyasha — COMMENTED** [review]
Add ticket link, please

**Review by VadymMakitra — APPROVED** [review]

**Review by NewArtMask — APPROVED** [review]

**Pull request merged**